docs/src/main/asciidoc/_developer-guide/Class_Loading_in_WildFly.adoc

```diff
@@ -450,3 +449,103 @@ relevant category to ERROR or higher.
+
+[source,options="nowrap"]
+----
+[domain@localhost:9990 /] /host=master/server=server-one/deployment=test-application.war:list-modules
+----
+
+[source,options="nowrap"]
+----
+[domain@localhost:9990 /] /host=master/server=server-one/deployment=test-application.ear/subdeployment=test-application.war:list-modules
+----
+
+By default, _list-modules_ operation shows the list of dependencies in a compact view, including only the module name. You can control this output using the attribute _verbose=[false*|true]_ to enabling/disabling a detailed response.
+
+The following output shows an example of a detailed view:
+
+[source,options="nowrap"]
+----
+[standalone@localhost:9990 /] /deployment=test-application.ear:list-modules(verbose=true)
+  {
+      "outcome" => "success",
+      "result" => {
+          "system-dependencies" => [
+              {
+                  "name" => "com.fasterxml.jackson.datatype.jackson-datatype-jdk8",
+                  "optional" => true,
+                  "export" => false,
+                  "import-services" => true
+              },
+              {
+                  "name" => "com.fasterxml.jackson.datatype.jackson-datatype-jsr310",
+                  "optional" => true,
+                  "export" => false,
+                  "import-services" => true
+              },
+              ...
+          ],
+          "local-dependencies" => [
+              {
+                "name" => "deployment.test-application.ear.test-application-ejb.jar",
+                "optional" => false,
+                "export" => false,
+                "import-services" => true
+              },
+              ...
+          ],
+          "user-dependencies" => [
+              {
+                  "name" => "com.fasterxml.jackson.datatype.jackson-datatype-jdk8",
+                  "optional" => false,
+                  "export" => false,
+                  "import-services" => false
+              },
+              {
+                  "name" => "org.hibernate:4.1",
+                  "optional" => false,
+                  "export" => false,
+                  "import-services" => false
+              },
+              ...
+          ]
+      }
+  }
+----
+
+
+The _list_modules_ command shows information in three different categories:
+
+* system-dependencies: These are the dependencies added implicitly by the server container.
+* local-dependencies: These are dependencies on other parts of the deployment.
+* user-dependencies: These are the dependencies defined by the user via manifest file or deployment-structure.xml.
+
+
+For each module, the following information is shown:
+
+* name: The module name and, if the slot name is not the default "main" slot, the slot name is concatenated after ":" character separator.
+* optional: If the dependency was added as an optional dependency.
+* export: If the dependency is being exported to other modules.
+* import-services: If this module is importing services from other dependencies.
```

Comment:
Is this more like

"If the module for the deployment or subdeployment is allowed to import services from the dependency."

?

That's the meaning of a 'services="import"' attribute on a dependency element in module.xml and looking at the setting of org.jboss.as.server.deployment.module.ModuleDependency.isImportServices() it is 'true' in the 'services="import"' situation, 'false' otherwise.

Reply:
Yes, "allowed to import" is a much better description.